### DISCUSSION TITLE
Should we drop support for ARM32 ?

### DISCUSSION THREAD

**Discussion by ajlennon** [DynamicDevices/meta-mono#234] [Polls]
I've had issues with builds recently code quantum safe code from the Open Quantum Project and now @zboszor is seeing this with meta-mono CI and the new GCC compiler.

On the one hand all the work we do here with Linux is ARM64 and ARM32 feels like it would be Zephyr or similar RTOS nowadays.

On the other hand I am reluctant to drop key legacy support.

What do people think?

**Comment by zboszor** [comment]
I have commented this in https://github.com/DynamicDevices/meta-mono/pull/231#issuecomment-2399771069 :

> I am not very familiar with ARM32 but it is possible that other subarchitectures have no problem, it's just the `cortexa15t2hf-neon-poky-linux-gnueabi` BSP has problems.
>
> FWIW, the ARM32 CI test can be easily revived when the GCC 14 build is fixed in OE-Core.

EDIT: This is just for styhead (Yocto 5.1) where [GCC 14.2 fails to build](https://github.com/DynamicDevices/meta-mono/actions/runs/11232177725/job/31223236463?pr=231#logs) for ARM32. As I wrote, it may be a temporary measure.

**Comment by CrazyGecko** [comment]
We're using the meta-mono for ARM32 currently in production and we will continue to need it in the future. It would be great if we find a way to continue support for ARM32.

**Reply by zboszor** [reply]
> We're using the meta-mono for ARM32 currently in production and we will continue to need it in the future. It would be great if we find a way to continue support for ARM32.

The reproducibility changes in master/styhead have nothing to do with the architecture.

GCC in Yocto 5.1 failed to build in the meta-mono CI. See https://github.com/DynamicDevices/meta-mono/pull/231#issuecomment-2399545781

This resulted in dropping ARM32 from CI in master and styhead and there's a [PR to revive it](https://github.com/DynamicDevices/meta-mono/pull/236) in case Yocto fixes the GCC build problem.

**Comment by ajlennon** [comment]
I think you emailed them didn't you @zboszor ? Did they come back with anything helpful?

**Reply by zboszor** [reply]
> I think you emailed them didn't you @zboszor ? Did they come back with anything helpful?

You were cc-ed on the [original post and the answers](https://lists.openembedded.org/g/openembedded-core/search?p=created%2C%2Ccortexa15t2hf-neon-poky-linux-gnueabi%2C20%2C2%2C0%2C0&q=fails%20to%20build%20for%20cortexa15t2hf-neon-poky-linux-gnueabi). There was nothing else on the matter.

**Comment by ajlennon** [comment]
Just so I'm clear on this until GCC build is fixed upstream Yocto doesn't support ARM32 ?
